src/compile/layout.ts

```diff
@@ -86,7 +86,7 @@ function parseUnitSizeLayout(model: UnitModel, channel: Channel): SizeComponent
 }
 
 function unitSizeExpr(model: UnitModel, channel: Channel, nonOrdinalSize: number): string {
-  if (model.has(channel) || (channel === X && model.has(X2)) || (channel === Y && model.has(Y2))) {
+  if (model.scale(channel) !== null || (channel === X && model.has(X2)) || (channel === Y && model.has(Y2))) {
```

Comment:
Wait, I mean I think you don't even need 
`|| (channel === X && model.has(X2)) || (channel === Y && model.has(Y2))`  here.